go-controller/pkg/ovn/gateway_init.go

```diff
@@ -625,8 +625,11 @@ func syncPolicyBasedRoutes(nodeName string, matches sets.String, priority, nexth
 	// sync and remove unknown policies for this node/priority
 	// also flag if desired policies are already found
 	for _, policyCompare := range policiesCompare {
-		if strings.Contains(policyCompare, fmt.Sprintf("%s\"", nodeName)) {
+		if strings.Contains(policyCompare, fmt.Sprintf("%s\"", nodeName)) &&
+			strings.Contains(policyCompare, fmt.Sprintf("%s.dst", l3Prefix)) {
```

Comment:
Sorry @fedepaol I think I led you down a bad path here. So this particular line wont work because it assumes the policy is matching on <prefix>.dst. A policy could match only on source (like on line 574) or no ip at all. I think we need to just stick with your original method checking if the foundNextHops is same family as nexthop. We already assume "nexthops" column only has one ip address, so we can do the same thing for now with your fix. Sorry again about the confusion.

Let's still keep your comment and also the fix where you removed the bad break.

Reply:
Ok